tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -946,11 +946,12 @@ decompress_chunk_path_create(PlannerInfo *root, CompressionInfo *info, int paral
 	path->cpath.flags = 0;
 	path->cpath.methods = &decompress_chunk_path_methods;
 
-	Assert(parallel_workers == 0 || compressed_path->parallel_safe);
-
-	path->cpath.path.parallel_aware = false;
-	path->cpath.path.parallel_safe = compressed_path->parallel_safe;
+	/* This path is only safe in a parallel plan if the Scan below
+	 * the DecompressChunk is parallel aware, otherwise the Scan
+	 * might produce duplicate tuples. */
+	path->cpath.path.parallel_safe = compressed_path->parallel_aware;
```

Comment:
The common practice I've seen is that this value is set from the `RelOptInfo` being planned, e.g., like this:

```
path->cpath.path.parallel_safe => compressed_rel->consider_parallel;
```

Sometimes also taking into account the parallel safeness of the subpath. So, I am wondering if we shouldn't make this call earlier when we create the `RelOptInfo` instead, so that it can be inherited consistently by all paths we consider. This seems very hard-coded to this particular path.

Reply:
safeness is not enough as can be seen by #2424

Reply:
You answer does not address my concern. 

Consider this: nowhere in the PostgreSQL code can I find a place where a path's `parallel_safe` setting is set based on the `parallel_aware` setting of another path; this should tell us something. Neither the code nor you PR description gives a good motivation or explanation for this setting and why it makes sense to deviate from the standard pattern seen elsewhere. 

The common pattern is as I described above, i.e., using `consider_parallel, which in most cases is propagated upwards by PostgreSQL to, e.g., upper and join rels automatically`. This seems logical since the relation being planned holds such information that gets applied to all paths considered for the planned relation.

Perhaps the setting of `consider_parallel` can take into account the parallel awareness of subnodes, but then that should happen at the relation level since there could be other, potentially parallel safe paths, that computes this relation (RelOptInfo).

So, my ask is twofold:

- Please provide an explanation why setting `parallel_safe` based on `parallel_aware` of the subpath makes sense in this particular case as an outlier.
- Please provide an explanation why we shouldn't make this decision primarily at the relation rather than the path level.

Reply:
All the fields on RelOptInfo apply to parallel and non-parallel path generation so they cannot be used for this test.